### PR TITLE
adding geometry_experimental

### PR DESCRIPTION
**Pull request by tfoote**
Connects to #153 

Requires #159 

Linux CI job passing: http://ci.ros2.org/job/ci_linux/677/#showFailuresLink

### CODE REVIEW

**Comment by esteve** [comment]
+1

**Comment by gerkey** [comment]
Is `geometry_experimental` also building successfully on Windows and OSX?

**Comment by tfoote** [comment]
@gerkey It's not tested on OSX or Windows. There's missing dependencies with console_bridge see #159 I don't know how we should proceed with console_bridge.

**Comment by gerkey** [comment]
Well, we have to figure something out so that we can get tf into ROS 2 for the upcoming alpha 3. I guess you should start by considering how to remove or replace `console_bridge` (maybe an alternative cross-platform implementation that just calls `printf()`/`cout()`?).

**Comment by wjwwood** [comment]
My opinion is that we should take `console_bridge` as it is in #133 and go with that rather than replacing it. I tested `console_bridge` and `class_loader` on all the platforms in my branch #133. The only outstanding thing to do (which could be delayed) is resolving the issue @dirk-thomas pointed out: https://github.com/ros2/ros2/pull/159#issuecomment-162706693

**Comment by wjwwood** [comment]
The only remaining issue with #133 is that it introduces warnings on Windows: https://github.com/ros2/ros2/pull/134#issuecomment-151649563

Those should be easy to fix, especially if you limit them to just `console_bridge` and don't take `class_loader` for now.

**Comment by tfoote** [comment]
This needs https://github.com/ros2/rclcpp/pull/172 to continue. I'll test that shortly.

**Comment by tfoote** [comment]
Linux CI passing: http://ci.ros2.org/job/ci_linux/721/
Clang is more strict on some datatype conversions inside sleep in rate.hpp, that need to be worked out: http://ci.ros2.org/job/ci_osx/599/console
And Windows there are some issues with implicit datatype conversions: http://ci.ros2.org/job/ci_windows_opensplice/785/console

**Comment by gerkey** [comment]
@wjwwood Please make a PR to allow review of the `ros2` branch of `console_bridge`.

@tfoote Same for `tf2_ament` branch of `geometry_experimental`.

**Comment by tfoote** [comment]
Please review the pull request with the changes here: https://github.com/ros/geometry_experimental/pull/134

File: ros2.repos [hunk 1]
```diff
@@ -106,7 +110,7 @@ repositories:
   ros2/rosidl:
     type: git
     url: https://github.com/ros2/rosidl.git
-    version: master
+    version: inject_rmw_dependenc
```
Comment (tfoote) [inline ros2.repos]: @wjwwood We added this here as a workaround to #174

**Comment by tfoote** [comment]
Final CI:
http://ci.ros2.org/job/ci_windows/863
http://ci.ros2.org/job/ci_linux/814/
http://ci.ros2.org/job/ci_osx/681/

**Comment by tfoote** [comment]
Final run of osx passed with only unrelated failures: http://ci.ros2.org/job/ci_osx/684/

**Comment by gerkey** [comment]
Those last couple of test failures should be addressed by ros2/system_tests#98.

**Comment by tfoote** [comment]
Squashed commit, upstream merged. Merging this.

**Pull request merged**